### DISCUSSION TITLE
Completion date

### DISCUSSION THREAD

**Discussion by pdxrlk** [mgmeyers/obsidian-kanban#220] [Ideas & Feature Requests]
I'd like to know the date I completed a task, so that I can assess my progress in weekly reviews, project reviews etc.

Some alternatives in order of my preference:

1. Provide an option "Set date upon task completion" which adds a date/time stamp (like archive) but does not remove it from the board.   Basically you can use the date/timestamp for either (a) task completion or (b) archive but not both.
2. Provide an option "Set date upon task completion" which would either set the date, or replace any existing date for the task
3. Provide an option "Archive upon task completion" which archives a task as soon as it is marked complete, thereby allowing the archive date to be used as task completion date.  But this removes it from the Kanban board :-(
4. Provide an option "Set date upon task completion" which creates an entirely new date (in addition to the "@" and archive dates.  Kind of a messy solution.

**Comment by MurrayMobile** [comment]
I would like to +1 the suggestion for a date time stamp when a card is completed. I would prefer it to be a separate datetime value for completion, with archive datetime also being tracked.

E.g., on a completed, archived item it might look something like:

> Archived 2022-09-20 19:09 // Completed 2022-09-20 19:06 // Details of issue here.

Thank you for this amazing plugin!
